src/libraries/System.Private.CoreLib/src/System/Environment.Windows.cs

```diff
@@ -83,6 +84,34 @@ private static string ExpandEnvironmentVariablesCore(string name)
             return builder.ToString();
         }
 
+        private static unsafe bool IsAdminProcess()
+        {
+            SafeTokenHandle? token = null;
+            try
+            {
+                if (Interop.Advapi32.OpenProcessToken(Interop.Kernel32.GetCurrentProcess(), (int)Interop.Advapi32.TOKEN_ACCESS_LEVELS.Read, out token))
+                {
+                    Interop.Advapi32.TOKEN_ELEVATION elevation = default;
+
+                    if (Interop.Advapi32.GetTokenInformation(
+                            token,
+                            Interop.Advapi32.TOKEN_INFORMATION_CLASS.TokenElevation,
+                            &elevation,
+                            (uint)sizeof(Interop.Advapi32.TOKEN_ELEVATION),
+                            out _))
+                    {
+                        return elevation.TokenIsElevated != Interop.BOOL.FALSE;
+                    }
+                }
+            }
+            finally
+            {
+                token?.Dispose();
+            }
+
+            throw Win32Marshal.GetExceptionForLastWin32Error();
```

Comment:
The error code can be overwritten by the Dispose call. `GetExceptionForLastWin32Error` has to be called right after the PInvoke that set the error.